### PR TITLE
enhancement: Turn test_script into a proper script [CORE-2933]

### PR DESCRIPTION
**Pull request by joelvdavies**
Replaces `test_scrip.txt` in the docs folder with a script `test_script.py` in a new folder named scripts.

This script is meant to be run manually using the non-admin1 test account. Running the script will run each command in turn requiring enter to be pressed to continue to the next one (for manual checking) or otherwise when running with --snapshot_overwrite it will generate snapshots of the output by saving them to a file. Then using --snapshot the next time will compare these against the previous, making note of any that are different. Any errored commands will be listed in the usual output after the script has finished.

Currently I get
image

 Some commands are commented out so they can't be run unless done manually:
- Upload - This can interfere with the get tests if not deleted again properly
- Delete - This id's will change every time making it harder to run unless extracting the id from a previous upload - which is possible but easy to break if the output from get changes

### CODE REVIEW

**Review by AJO-Python — COMMENTED** [review]
One small comment about the `DAFNI_CLI_SCRIPT` and 'SNAPSHOT_SAVE_LOCATION` vars default values

File: scripts/test_script.py [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+from pathlib import Path
+from typing import Optional, Union
+
+import click
+
+# Not sure why it wasn't finding this, but this will do for now
+DAFNI_CLI_SCRIPT = "/home/joel/.local/bin/dafni"
+
+# Where to save snapshots
+SNAPSHOT_SAVE_LOCATION = "/home/joel/dafni_cli_snapshots/"
```
[the remark this answers is not in the document]
Comment (joelvdavies) [reply scripts/test_script.py]: Yeah environments variables sounds like a good idea anyway, I have modified it to use them if defined or fall back to more generic paths from the home directory.

File: scripts/test_script.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+import click
+
+home_dir = Path.home()
+
+# The cli executable is
+DAFNI_CLI_SCRIPT = os.getenv("DAFNI_CLI_SCRIPT") or f"{home_dir}/.local/bin/dafni"
+
+# Where to save snapshots
+DAFNI_SNAPSHOT_SAVE_LOCATION = (
```
Comment (sarahbyrnie) [inline scripts/test_script.py:28]: Maybe state somewhere that this needs to be an already existing folder

**Review by sarahbyrnie — COMMENTED** [review]
Could you add somewhere in the README or notes on usage how you run the subsets of tests? I had to use the --help on the command line to figure it out. Other than that it all looks good!

File: scripts/test_script.py (hunk 2 above)
Comment (joelvdavies) [reply scripts/test_script.py:28]: Put it in the comment and in the dev notes.

**Review by sarahbyrnie — APPROVED** [review]

**Pull request merged**